src/org/sosy_lab/java_smt/solvers/bitwuzla/BitwuzlaFloatingPointManager.java

```diff
@@ -0,0 +1,355 @@
+    return termManager.mk_fp_nan(mkFpaSort(pType));
+  }
+
+  @Override
+  protected Term castToImpl(
+      Term pNumber, boolean pSigned, FormulaType<?> pTargetType, Term pRoundingMode) {
+    if (pTargetType.isFloatingPointType()) {
+      FormulaType.FloatingPointType targetType = (FormulaType.FloatingPointType) pTargetType;
+      return termManager.mk_term(
+          Kind.FP_TO_FP_FROM_FP,
+          pRoundingMode,
+          pNumber,
+          targetType.getExponentSize(),
+          targetType.getMantissaSize() + 1);
+    } else if (pTargetType.isBitvectorType()) {
+      FormulaType.BitvectorType targetType = (FormulaType.BitvectorType) pTargetType;
+      if (pSigned) {
+        return termManager.mk_term(Kind.FP_TO_SBV, pRoundingMode, pNumber, targetType.getSize());
+      } else {
+        return termManager.mk_term(Kind.FP_TO_UBV, pRoundingMode, pNumber, targetType.getSize());
+      }
+    } else {
+      throw new UnsupportedOperationException(
+          "Attempted cast of FP to an unsupported type: " + pTargetType + ".");
+    }
+  }
+
+  @Override
+  protected Term castFromImpl(
+      Term pNumber, boolean pSigned, FloatingPointType pTargetType, Term pRoundingMode) {
+    FormulaType<?> formulaType = getFormulaCreator().getFormulaType(pNumber);
+    if (formulaType.isFloatingPointType()) {
+      return castToImpl(pNumber, pSigned, pTargetType, pRoundingMode);
+    } else if (formulaType.isBitvectorType()) {
+      if (pSigned) {
+        return termManager.mk_term(
+            Kind.FP_TO_FP_FROM_SBV,
+            roundingMode,
+            pNumber,
+            pTargetType.getExponentSize(),
+            pTargetType.getMantissaSize() + 1);
+      } else {
+        return termManager.mk_term(
+            Kind.FP_TO_FP_FROM_UBV,
+            roundingMode,
+            pNumber,
+            pTargetType.getExponentSize(),
+            pTargetType.getMantissaSize() + 1);
+      }
+
+    } else {
+      throw new UnsupportedOperationException(
+          "Attempted cast towards FP from an unsupported type" + ": " + formulaType + ".");
+    }
+  }
+
+  @Override
+  protected Term fromIeeeBitvectorImpl(Term pNumber, FloatingPointType pTargetType) {
+    return termManager.mk_term(
+        Kind.FP_TO_FP_FROM_BV,
+        pNumber,
+        pTargetType.getExponentSize(),
+        pTargetType.getMantissaSize() + 1);
+  }
+
+  @Override
+  protected Term toIeeeBitvectorImpl(Term pNumber) {
+    // FIXME: We should use a reserved symbol for the fresh variables
+    int sizeExp = pNumber.sort().fp_exp_size();
+    int sizeSig = pNumber.sort().fp_sig_size();
+
+    Sort bvSort = termManager.mk_bv_sort(sizeExp + sizeSig);
+
+    // Note that NaN is handled as a special case in this method. This is not strictly necessary,
+    // but if we just use "fpTerm = to_fp(bvVar)" the NaN will be given a random payload (and
+    // sign). Since NaN payloads are not preserved here anyway we might as well pick a canonical
+    // representation.
```

Comment:
I did ask the Bitwuzla developers about this method and [this](https://github.com/bitwuzla/bitwuzla/discussions/103#discussioncomment-8339056) is the response I got. Then again I do remember seeing one false positive when running CPAchecker on the sv-comp set with the workaround enabled. We could disable it for now, but this will break a lot of other programs from the sv-comp set that need to cast from float to bitvector.